addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -61,6 +61,8 @@ def send_mail_test(self):
                 'attachment_ids': [(4, attachment.id) for attachment in mailing.attachment_ids],
                 'auto_delete': False,  # they are manually deleted after notifying the document
                 'mail_server_id': mailing.mail_server_id.id,
+                'res_id': self.env.user.id,
+                'model': 'res.users'
```

Comment:
And if you click on the link, aren't you going to blacklist your own user :o ?

Reply:
It is possible for the user to blacklist himself, but from what I tested it is not done automatically.
You can blacklist yourself only if you press on the blacklist button yourself.
The only action done by opening the unsubscribe view is that you are removed from the mailing list configured, which can be reverted in the same view.

Reply:
That's the case if the mailing is linked to a mailing list, but otherwise it should blacklist the email (code is different depending on mailing list targetting mailing lists or 'business documents').

Reply:
I just pushed another version for this in a second commit.
In this one we don't use the res_id and model but I modified the condition inside `_prepare_outgoing_list` so that if res_id is not set we don't stop the execution. 
This gives the following behavior:
 * Links are correctly converted and gives us the correct online view
 * The unsubscribe button redirects to a 403 Forbidden.
This is possible because the res_id is only used generate tokens and encode urls with werkzeug.

Now we don't have to worry about the user blacklisting himself.

Do you prefer this version or the previous one?

Reply:
Discussed this with @std-odoo, and using no res_id isn't a great idea as in later versions the res_id is important to check if an access token is valid or not.

So what's possible is that I can set a context key so that the unsubscribe links aren't converted and only the view online works.
This way we have a single fix for all versions.

Reply:
Note that starting from 17, it can just be replaced by a link to mailing/my (don't remember the actual route, stuck in intermediate version). It can benefit from improved routes.

For previous version, not sure what is best. We don't have "easy route for unsubscribe" anyway, either you have the original one, either a crashing one, either nothing. Meh.

Side note: why a fix in 16.3 ?

Reply:
Because when I tested the feature in 16 it already worked and when I tested in the lowest sub version accessible from 16 (16.3) the issue arose. So I decided to fix it in this sub version.

Side-note: in 16 the method doesn't check if we have a res_id set when preparing the outgoing list. => Same behavior as I explained 2 comments earlier.